libass/ass.c

```diff
@@ -1455,18 +1455,21 @@ char *ass_load_file(ASS_Library *library, const char *fname, FileNameSource hint
         return NULL;
     }
 
-    const size_t CHUNK_SIZE = 16 * 1024;
-    size_t bufmax = 128 * CHUNK_SIZE;
+    size_t bufmax = 2 * 1024 * 1024;
+    ass_get_file_size(fp, &bufmax);
+    if (bufmax > SIZE_MAX - 1)
+        return NULL;
```

Comment:
You may want to short-circuit if `bufmax` is zero. Failing is probably fine, as there’s nothing useful in the file anyway.

Reply:
good idea, done